### DISCUSSION TITLE
Add support for PostgreSQL data type macaddr8

### DISCUSSION THREAD

**Discussion by daniele-solutiontech** [diesel-rs/diesel#4312] [Ideas]
Diesel now supports PostgreSQL data type `macaddr` (EUI-48), it would be awesome to support also the `macaddr8` (EUI-64)

**Comment by weiznich** [comment]
We are happy to accept a PR that adds support for this type.

**Comment by daniele-solutiontech** [comment]
I’m currently working on adding macaddr8 support for a pull request, but I’m not entirely clear on the purpose of the ByteArrayProxy code. I’ve been reviewing the implementation for macaddr6 and came across the following:
```
#[allow(dead_code)]
mod foreign_derives {
    use super::*;
    use crate::deserialize::FromSqlRow;
    use crate::expression::AsExpression;

    #[derive(AsExpression, FromSqlRow)]
    #[diesel(foreign_derive)]
    #[diesel(sql_type = MacAddr)]
    struct ByteArrayProxy([u8; 6]);
}
```
Apologies—I’m still relatively new to the Diesel library and could use some guidance.

**Reply by weiznich** [reply]
That's one part of the necessary trait setup to convert a `MacAddr` to/from a `[u8;6]` on rust side. We likely want something similar for the new `MacAddr8` type. The other parts are the `FromSql`/`ToSql` impls below. 

I think it should be possible to essentially just copy all of that + change the array size to 8 + use the `MacAddr8` type there (needs to be created).

**Reply by daniele-solutiontech** [reply]
I was considering implementing something like this:
```
#[allow(dead_code)]
mod foreign_derives {
    use super::*;
    use crate::deserialize::FromSqlRow;
    use crate::expression::AsExpression;

    #[derive(AsExpression, FromSqlRow)]
    #[diesel(foreign_derive)]
    #[diesel(sql_type = MacAddr)]
    struct MacAddressProxy(MacAddress);

    #[derive(Debug, PartialEq)]
    enum MacAddress {
        MacAddr6([u8; 6]),
        MacAddr8([u8; 8]),
    }
}
```
Afterward, I would simply replicate the ToSql and FromSql implementations. What do you think about it?

**Reply by weiznich** [reply]
That would be a breaking change for the old impl. Just replicating everything for the new type, including the byte wrapper. 

Also to just write that out explicitly: We also need a distinct MacAddr8 type for representing the SQL side type. You also can copy the definition of the existing type there, but you need to adjust the oids.

**Reply by daniele-solutiontech** [reply]
I’ve just submitted a PR. Please let me know if anything is incorrect or missing.
